tensorflow/core/kernels/cwise_ops.h

```diff
@@ -43,6 +43,39 @@ struct functor_traits<scalar_fmod2_op<T>> {
   };
 };
 
+template<typename T> struct scalar_asinh_op {
+  EIGEN_EMPTY_STRUCT_CTOR(scalar_asinh_op)
+  EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE const T operator()(const T& a) const {
+    return std::asinh(a);
+  }
+};
+template<typename T>
+struct functor_traits<scalar_asinh_op<T>> {
+  enum { Cost = 5 * NumTraits<T>::MulCost, PacketAccess = false };
+};
+
+template<typename T> struct scalar_acosh_op {
+  EIGEN_EMPTY_STRUCT_CTOR(scalar_acosh_op)
+  EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE const T operator()(const T& a) const {
+    return std::acosh(a);
```

Comment:
It's surprising that everything works fine on Linux and fails on Windows. I don't think this is the case but could it have something to do with the standard library implementation of acosh, since that is what is used in the background for computation.